source/gui/guiHelper.py

```diff
@@ -363,7 +362,20 @@ def addDialogDismissButtons(self, buttons, separated=False):
 		self.dialogDismissButtonsAdded = True
 		return buttons
 
+
+def safeAppExit():
+	def ensureMainLoopExited():
+		if wx.GetApp().GetMainLoop():
+			wx.GetApp().ExitMainLoop()
+
+	if not wx.GetApp().GetMainLoop():
+		log.warning("app trying to exit after already exited")
+		return
+	if not wx.GetApp().GetTopWindow().IsBeingDeleted():
+		return wx.CallAfter(wx.GetApp().GetTopWindow().Destroy)

```

Comment:
Can you please include a comment explaining why the destroy must be done with a CallAfter? I'm assuming one reason is to ensure it is executed from NVDA's main (GUI) thread, as safeExit could be called from anywhere or any thread.

Reply:
yep that is the reason, added some comments